### PR TITLE
RELATED: RAIL-2107 - Add sdk-embedding and utilities packages

### PR DESCRIPTION
**Pull request by lupko**
### sdk-embedding

contains /embedding part of the gooddata typings + post message utilities from js-utils:

-  Move export definitions from client bear to model-bear
-  Fixup all TSDoc / api-extractor problems
-  Move post-events from js-utils

### utilities

low-level utility code previously located in js-utils. moved to SDK to reduce dependencies on GD packages outside of the monorepo. moved just the code the SDK needs.

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/217/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1809/ : FAILURE in 9m 14s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/230/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1819/ : FAILURE in 4m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/231/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1820/ : SUCCESS in 9m 05s

File: libs/utilities/src/tests/stringUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// (C) 2007-2020 GoodData Corporation
+import { randomString, shortenText, simplifyText, parseStringToArray, hashCodeString } from "../stringUtils";
+
+describe("#randomString", () => {
```
Comment (no23reason) [inline libs/utilities/src/tests/stringUtils.test.ts:4]: Nitpick: we do not use the `#something` notation anywhere else. Should we remove it here?

File: libs/utilities/src/tests/stringUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        expect(simplifyText(123)).toEqual("123");
+        expect(simplifyText(" - ")).toEqual("___");
+        expect(simplifyText("123456")).toEqual("123456");
+        expect(simplifyText("Too long")).toEqual("too_long");
+        expect(simplifyText(null)).toEqual("");
+    });
+});
+
+describe("parseStringToArray", () => {
+    it("should parse string to array correctly", () => {
```
Comment (no23reason) [inline libs/utilities/src/tests/stringUtils.test.ts:39]: This should probably be a parametric test.

File: libs/utilities/src/tests/stringUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    it("should NOT shorten text when text is shorter than maxLength", () => {
+        const shortened = shortenText("Lorem ipsum dolor sit amet", { maxLength: 100 });
+        expect(shortened).toEqual("Lorem ipsum dolor sit amet");
+    });
+});
+
+describe("#simplifyText", () => {
+    it("should simplify text", () => {
```
Comment (no23reason) [inline libs/utilities/src/tests/stringUtils.test.ts:28]: This should probably be a parametric test.

File: libs/utilities/src/tests/stringUtils.test.ts (hunk 1 above)
Comment (lupko) [reply libs/utilities/src/tests/stringUtils.test.ts]: fixed

File: libs/utilities/src/tests/stringUtils.test.ts (hunk 3 above)
Comment (lupko) [reply libs/utilities/src/tests/stringUtils.test.ts]: fixed

File: libs/utilities/src/tests/stringUtils.test.ts (hunk 2 above)
Comment (lupko) [reply libs/utilities/src/tests/stringUtils.test.ts]: fixed

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/234/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1823/ : SUCCESS in 14m 56s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/210/ : FAILURE in 4m 30s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/465/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1825/ : SUCCESS in 10m 32s

**Pull request merged**